android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -797,4 +802,20 @@ private String encodePresenceAction(PresenceMessage.Action action) {
     return jsonMap;
   }
 
+  private Map<String, Object> encodeRemoteMessage(RemoteMessage message) {
```

Comment:
I did create an issue to address this in the future: https://github.com/ably/ably-flutter/issues/174 This requires some thought to design a class which contains data from iOS and Android. Also, it might be challenging (but possible) to get some fields on both platforms, whereas some are impossible (e.g. priority on Android vs. iOS).

I didn't document that we are only passing certain fields of an **FCM** RemoteMessage, primarily because on iOS there is **very little** to give anyway: its just an NSDictionary. An ably remote message is not the same as an FCM RemoteMessage, so I didn't think it was worth mentioning we lack some FCM RemoteMessage fields.

IMHO the MVP would be this RemoteMessage, the notification and the data map.

Reply:
Again, not completely sure about the details but if there is an exception / exclusion from spec, it should be documented. Maybe we can quickly discuss the design of this one as well?

Reply:
I don't think it's related to the feature spec or exceptions in this case though 😅 :)